segment/val.py

```diff
@@ -461,7 +462,7 @@ def main(opt):
                     r, _, t = run(**vars(opt), plots=False)
                     y.append(r + t)  # results and times
                 np.savetxt(f, y, fmt='%10.4g')  # save
-            os.system('zip -r study.zip study_*.txt')
+            subprocess.run('zip -r study.zip study_*.txt'.split())
```

Comment:
I don't think this change really makes anything more secure.

(This repeats for the other zip.)